src/matching/Apply_equivalences.ml

```diff
@@ -45,30 +45,29 @@ let match_e_e_for_equivalences _ruleid lang a b =
 let subst_e (env : Env.t) e =
   let bindings = env.full_env in
   let visitor =
-    M.mk_visitor
-      {
-        M.default_visitor with
-        M.kexpr =
-          (fun (k, _) x ->
-            match x.e with
-            | N (Id ((str, _tok), _id_info)) when MV.is_metavar_name str -> (
-                match List.assoc_opt str bindings with
-                | Some (MV.Id (id, Some idinfo)) ->
-                    (* less: abstract-line? *)
-                    N (Id (id, idinfo)) |> G.e
-                | Some (MV.E e) ->
-                    (* less: abstract-line? *)
-                    e
-                | Some _ ->
-                    failwith
-                      (spf "incompatible metavar: %s, was expecting an expr" str)
-                | None ->
-                    failwith
-                      (spf "could not find metavariable %s in environment" str))
-            | __else__ -> k x);
-      }
+    object (_self : 'self)
+      inherit [_] AST_generic.map_legacy as super
```

Comment:
What is map_legacy? what is the difference with the default mapper?
Maybe worth adding a comment here?